kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/usb/UsbModemDevice.java

```diff
@@ -197,4 +248,87 @@ private int getDigitPosition(String devName) {
             return pos;
         }
     }
+
+    private class TtyDev {
+
+        private final String portName;
+        private Integer interfaceNumber;
+
+        public TtyDev(String portName) {
+            this.portName = portName;
+        }
+
+        public TtyDev(String portName, Integer interfaceNumber) {
+            this.portName = portName;
+            this.interfaceNumber = interfaceNumber;
+        }
+
+        public String getPortName() {
+            return this.portName;
+        }
+
+        public Integer getInterfaceNumber() {
+            return this.interfaceNumber;
+        }
+
+        private UsbModemDevice getOuterType() {
+            return UsbModemDevice.this;
+        }
+
+        @Override
+        public String toString() {
+            return "TtyDev [portName=" + this.portName + ", interfaceNumber=" + this.interfaceNumber + "]";
```

Comment:
NPE if interface number is null

Reply:
@cdealti If we decide that the interface number can be null (and this is the case for this fix), we have to take care of this NPE.